index/scorch/segment/zap/segment_test.go

```diff
@@ -515,3 +518,83 @@ func TestOpenMultiWithTwoChunks(t *testing.T) {
 		t.Errorf("expected count to be 1, got %d", count)
 	}
 }
+
+func TestSegmentVisitableDocValueFieldsList(t *testing.T) {
+	_ = os.RemoveAll("/tmp/scorch.zap")
+
+	memSegment := buildMemSegmentMulti()
+	err := PersistSegment(memSegment, "/tmp/scorch.zap", 1)
+	if err != nil {
+		t.Fatalf("error persisting segment: %v", err)
+	}
+
+	seg, err := Open("/tmp/scorch.zap")
+	if err != nil {
+		t.Fatalf("error opening segment: %v", err)
+	}
+
+	cerr := seg.Close()
```

Comment:
why do we close the segment we just opened, and then keep using it in the next block?

edit: i see what happened, you copy/pasted it from a defer and just removed the deferred part, but now that code is the wrong place (defer executes out of the sequential flow of normal code)

Reply:
Yikes...Its a total copy pasted carelessness error.. :(